pkg/util/util.go

```diff
@@ -239,3 +244,156 @@ func SplitQualifiedName(str string) (string, string) {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[2])
+		}
+		route.Gateway = ip
+	}
+	return routes, nil
+}
+
+func parseIP(str string) (net.IP, error) {
+	bytes, err := hex.DecodeString(str)
+	if err != nil {
+		return nil, err
+	}
+	if len(bytes) != net.IPv4len {
+		return nil, fmt.Errorf("only IPv4 is supported")
+	}
+	bytes[0], bytes[1], bytes[2], bytes[3] = bytes[3], bytes[2], bytes[1], bytes[0]
+	return net.IP(bytes), nil
+}
+
+func getRouteTables() (string, error) {
+	args := []string{"/proc/net/route"}
+	out, err := exec.Command("cat", args...).Output()
+	if err != nil {
+		return "", err
+	}
+	return string(out), err
+}
+
+func isInterfaceUp(intf *net.Interface) bool {
+	if intf.Flags&net.FlagUp != 0 {
+		glog.V(4).Infof("Interface %v is up\n", intf.Name)
+		return true
+	}
+	return false
+}
+
+func getFinalIP(addrs []net.Addr) (net.IP, error) {
+	if len(addrs) > 0 {
+		for i := range addrs {
+			glog.V(4).Infof("Checking addr  %s.\n", addrs[i].String())
+			ip, _, err := net.ParseCIDR(addrs[i].String())
+			if err != nil {
+				return nil, err
+			}
+			//Only IPv4
+			if ip.To4() != nil {
+				if !ip.IsLoopback() {
+					glog.V(4).Infof("IP found %v\n", ip)
+					return ip, nil
+				} else {
+					glog.V(4).Infof("Loopback found %v\n", ip)
+				}
+			} else {
+				glog.V(4).Infof("%v is not a valid IPv4 address\n", ip)
+			}
+
+		}
+	}
+	return nil, nil
+}
+
+func getIPFromInterface(intfName string) (net.IP, bool, error) {
+	intf, err := net.InterfaceByName(intfName)
+	if err != nil {
+		glog.Errorf("Can't find an interface named %q: %v\n", intfName, err)
+		return nil, false, err
+	}
+	if isInterfaceUp(intf) {
+		addrs, err := intf.Addrs()
+		if err != nil {
+			glog.Errorf("Can't get addresses for interface %q: %v\n", intfName, err)
+			return nil, false, err
+		}
+		glog.V(4).Infof("Interface %q has %d addresses :%v.\n", intfName, len(addrs), addrs)
+		finalIP, err := getFinalIP(addrs)
+		if err != nil {
+			glog.Errorf("Can't get IP from addresses of interface %q: %v\n", intfName, err)
+			return nil, false, err
+		}
+		if finalIP != nil {
+			glog.V(4).Infof("valid IPv4 address for interface %q found as %v.\n", intfName, finalIP)
+			return finalIP, true, nil
+		}
+	}
+
+	return nil, false, nil
+}
+
+func ChooseHostInterface() (net.IP, bool, error) {
+	zero := net.IP{0, 0, 0, 0}
+	routedata, err := getRouteTables()
+	if err != nil {
+		glog.Errorf("getRouteTables, %v\n", err)
+		return nil, false, err
+	}
+	r := strings.NewReader(routedata)
+
+	routes, err := getRoutes(r)
+	if err != nil {
+		glog.Errorf("getRoutes, %v\n", err)
+		return nil, false, err
+	}
+	for i := range routes {
+		//find interface with gateway
+		if routes[i].Destination.Equal(zero) {
+			glog.V(4).Infof("Default route transits interface %q\n", routes[i].Interface)
+			ip, found, err := getIPFromInterface(routes[i].Interface)
+			if err != nil {
+
+				return nil, false, err
+			}
+			if found {
+				glog.Infof("Choosing IP %v \n", ip)
```

Comment:
don't log "normal" stuff in here - all logs should be either V(3) or higher or "wow, that was unexpected!" stuff.